src/lib/file_operations.js

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env node
+/* eslint-disable no-console */
+
+const fs = require("fs")
+const { generateMarkdownFromJson } = require("./markdown_generators")
+const cliProgress = require("cli-progress")
+const progressBar = new cliProgress.SingleBar(
+  {},
+  cliProgress.Presets.shades_classic
+)
+let totalDirectories = 0
+let directoriesScanned = 0
+let filesProcessed = 0
+
+const scanCourses = (source, destination) => {
+  /*
+    This function scans the source directory for course folders
+  */
+  fs.readdir(source, (err, directories) => {
+    // Count the total amount of directories and start the progress bar
+    directories.forEach(directory => {
+      const coursePath = source + directory
```

Comment:
We should use [path.join](https://nodejs.org/docs/latest-v10.x/api/path.html#path_path_join_paths) instead of string concatenation for concatenating paths together.